packages/flutter_tools/lib/src/build_system/targets/android.dart

```diff
@@ -247,6 +247,10 @@ class AndroidAot extends AotElfBase {
       extraGenSnapshotOptions.add('--trace-precompiler-to=${precompilerTraceFile.path}');
     }
 
+    // AGP handles stripping debug symbols from native libraries, so we default
+    // to not stripping on Android.
+    extraGenSnapshotOptions.add('--no-strip');
```

Comment:
I wonder if this decision should happen on the layer below: e.g. `build` has Apple specific logic for producing `dSYM`s already, so maybe that's the place for Android target as well e.g. if targeting Android don't strip by default? Or are there codepath where we build for Android but not via AGP?